sdk/appconfiguration/Azure.Data.AppConfiguration/src/CreateSnapshotOperation.cs

```diff
@@ -0,0 +1,141 @@
+        }
+
+        /// <inheritdoc/>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default) => UpdateStatusAsync(false).EnsureCompleted();
+
+        /// <inheritdoc/>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) => await UpdateStatusAsync(true).ConfigureAwait(false);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<ConfigurationSettingsSnapshot>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(s_defaultPollingInterval, cancellationToken);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<ConfigurationSettingsSnapshot>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
+            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        private async ValueTask<Response> UpdateStatusAsync(bool async)
+        {
+            if (!_hasCompleted)
+            {
+                using DiagnosticScope? scope = _diagnostics?.CreateScope($"{nameof(CreateSnapshotOperation)}.{nameof(UpdateStatus)}");
+                scope?.Start();
+
+                try
+                {
+                    // Get the latest status
+                    Response update = async
+                        ? await _client.GetOperationDetailsAsync(_snapshotName).ConfigureAwait(false)
+                        : _client.GetOperationDetails(_snapshotName);
+
+                    JsonElement result = JsonDocument.Parse(update.ContentStream).RootElement;
+                    var status = result.GetProperty("status");
+
+                    // Check if the operation is no longer running
+                    _hasCompleted = IsJobComplete(status.ToString());
+                    if (_hasCompleted)
+                    {
+                        // Only want to set this if the operation has completed.
+                        // The service does not include any snapshot information in its operation details, so
+                        // request them manually.
+                        ConfigurationSettingsSnapshot updatedSnapshot = async
+                            ? await _client.GetSnapshotAsync(_snapshotName).ConfigureAwait(false)
+                            : _client.GetSnapshot(_snapshotName);
+
+                        _snapshot = updatedSnapshot;
+
+                        _hasValue = true;
+                    }
+
+                    // Update raw response
+                    _rawResponse = update;
+                }
+                catch (Exception e)
+                {
+                    scope?.Failed(e);
+                    throw;
+                }
+            }
+
+            return GetRawResponse();
+        }
+
+        private static bool IsJobComplete(string status)
+        {
+            if (status == "Succeeded" || status == "Failed" || status == "Canceled")
```

Comment:
Instead of hardcoding the status value here, can we defined an enum same as defined in the swagger [here](https://github.com/Azure/azure-rest-api-specs/blob/48125b4a8cd39b3034da70bc57d6ddc738b10de5/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/preview/2022-11-01-preview/appconfiguration.json#L1780-L1786)?